src/H5Faccum.c

```diff
@@ -879,6 +879,8 @@ H5F__accum_free(H5F_shared_t *f_sh, H5FD_mem_t H5_ATTR_UNUSED type, haddr_t addr
 
                 /* Calculate the size of the overlap with the accumulator, etc. */
                 H5_CHECKED_ASSIGN(overlap_size, size_t, (addr + size) - accum->loc, haddr_t);
+                if (overlap_size > accum->size)
```

Comment:
I'm still not sure we've found the root of the problem. The condition to get into this block is effectively `addr+size < accum->loc + accum->size`. Therefore `overlap_size`, which is `addr+size-accum->loc` should be less than `accum->size` undern normal conditions. Is one of the quantities involved (addr, size, accum->loc, accum->size) very large and causing wraparound? If so, is there a real reason it should be this way, or can we detect an impossibly large value when it's loaded from disk? We should also add H5_CHECK_OVERFLOW and similar checks as appropriate.